src/CounterFileRepository.php

```diff
@@ -39,8 +39,20 @@ public function __construct(string $storagePath)
     public function getViewsCountByUsername(Username $username): int
     {
         $counterFilePath = $this->getCounterFilePath($username);
+        $count=file_exists($counterFilePath) ? (int) file_get_contents($counterFilePath) : 0;
+        $abbreviate= $_GET['abbreviate'] ?? false;
+        function abbreviateNumber($value) {
```

Comment:
Pull out transformation from this class, it should not modify the value. If you want to make a transformation, make it as closer to the output (rendering) as possible.

One more thing to consider. There is CounterPdoRepository class, and if we will move transformation to the badge rendering, then it will work for database counter too.